### ISSUE TITLE
Can't deploy with prisma

### ISSUE THREAD

**Issue by ryanflorence** [superfly/flyctl#405]
I'm not sure where the issue lies between prisma and fly.io, so I've opened issues in both repos hoping somebody has a clue about what might be happening 🥴

Prisma has a postinstall script that runs after `npm install`, for whatever reason it's hitting an EOF error (and I don't know which file).

Here's a little repo to reproduce: https://github.com/ryanflorence/fly-prisma

The app runs fine locally, here's the output from the deploy after Node.js is installed and it runs `npm install`:

```
[Parsing package.json]
[INFO] Parsing package.json
[INFO] Using npm v6.14.12 from Node
[INFO] Installing node modules from ./package-lock.json

> prisma@2.21.2 preinstall /workspace/node_modules/prisma
> node scripts/preinstall-entry.js


> @prisma/engines@2.21.0-36.e421996c87d5f3c8f7eeadd502d4ad402c89464d postinstall /workspace/node_modules/@prisma/engines
> node download/index.js


> prisma@2.21.2 install /workspace/node_modules/prisma
> node scripts/install-entry.js


> @prisma/client@2.21.2 postinstall /workspace/node_modules/@prisma/client
> node scripts/postinstall.js

Prisma schema loaded from prisma/schema.prisma
Error executing lifecycle: unexpected EOF
```

## `fly.toml`

My fly.toml is the default from `fly launch`

```toml
# fly.toml file generated for solitary-wave-5918 on 2021-04-25T19:56:40-06:00

app = "..."

kill_signal = "SIGINT"
kill_timeout = 5

[build]
  builder = "heroku/buildpacks:20"

[env]
  PORT = "8080"

[experimental]
  auto_rollback = true

[[services]]
  http_checks = []
  internal_port = 8080
  protocol = "tcp"
  script_checks = []

  [services.concurrency]
    hard_limit = 25
    soft_limit = 20
    type = "connections"

  [[services.ports]]
    handlers = ["http"]
    port = 80

  [[services.ports]]
    handlers = ["tls", "http"]
    port = 443

  [[services.tcp_checks]]
    grace_period = "1s"
    interval = "15s"
    restart_limit = 6
    timeout = "2s"

```

 [Here's the issue on prisma's repo](https://github.com/prisma/prisma/issues/6779)

**Comment by mrkurt** [comment]
This looks like a problem with the remote builder (#404), if I run with a local Docker daemon it seems to build just fine.

**Comment by mrkurt** [comment]
It's most likely an issue with our remote builder breaking the network connection. I don't think it's an EOF generated from node, but a network level eof. If you run a local Docker daemon it will probably work.

**Comment by janpio** [comment]
I am debugging this on the @Prisma side 👋 

This happens in a part of our `postinstall` script, which I could skip with setting `PRISMA_SKIP_POSTINSTALL_GENERATE=true` env var (to see if the rest of the process would work fine). Setting that in the fly.toml or via secret does not seem to influence the environment when this package installation (`npm install`) is run though. Is there a way to set that for this step?

**Comment by mrkurt** [comment]
You can pass that as a build arg: `fly deploy --build-arg PRISMA_SKIP_POSTINSTALL_GENERATE=true`.

**Comment by janpio** [comment]
Ok, setting this definitely unblock _this_ problem:

```
[...]

> prisma@2.21.2 install /workspace/node_modules/prisma
> node scripts/install-entry.js


> @prisma/client@2.21.2 postinstall /workspace/node_modules/@prisma/client
> node scripts/postinstall.js

added 64 packages in 3.726s

[Warning: Skip pruning because NODE_ENV is not 'production'.]

[INFO] Discovering process types
[INFO] Procfile declares types -> (none)
===> EXPORTING

[...]
```

Now we'll see if this pops up again (which I kinda expect - or the app just does not work without the generated Prisma Client).

And then of course we can figure out what is happening here and why - and how to fix it.

**Comment by janpio** [comment]
Setting `DEBUG=*` for `npx prisma generate` gives a bit more output what it is doing under the hood:
```
> script@undefined postinstall /workspace
> npx prisma generate

2021-04-26T14:02:14.396Z prisma:loadEnv project root found at /workspace/package.json
2021-04-26T14:02:14.405Z prisma:tryLoadEnv Environment variables not found at null
2021-04-26T14:02:14.405Z prisma:tryLoadEnv Environment variables loaded from ./prisma/.env
[dotenv][DEBUG] did not match key and value when parsing line 2:
[dotenv][DEBUG] did not match key and value when parsing line 3:
Environment variables loaded from prisma/.env
2021-04-26T14:02:14.409Z prisma:engines using NAPI: false
2021-04-26T14:02:14.409Z prisma:engines binaries to download query-engine, migration-engine, introspection-engine, prisma-fmt
Prisma schema loaded from prisma/schema.prisma
2021-04-26T14:02:14.823Z prisma:download copying /home/heroku/.cache/prisma/master/e421996c87d5f3c8f7eeadd502d4ad402c89464d/debian-openssl-1.1.x/query-engine to /workspace/node_modules/prisma/query-engine-debian-openssl-1.1.x
Error executing lifecycle: unexpected EOF
```

So this seems to be related to our binary downloading and extracting and moving to the expected location.

**Comment by ryanflorence** [comment]
Thanks for the debug tip:

Mine get's hung up on this last line, I'm going on like 10 minutes waiting for it:

```
> @prisma/client@2.21.2 postinstall /workspace/node_modules/@prisma/client
> node scripts/postinstall.js

// ... other stuff

Prisma schema loaded from prisma/schema.prisma
2021-04-26T14:44:06.848Z prisma:download copying /home/heroku/.cache/prisma/master/e421996c87d5f3c8f7eeadd502d4ad402c89464d/debian-openssl-1.1.x/query-engine to /workspace/node_modules/prisma/query-engine-debian-openssl-1.1.x
```

**Comment by janpio** [comment]
Yes, that is the one that then probably fails with that error:

```
2021-04-26T14:02:14.823Z prisma:download copying /home/heroku/.cache/prisma/master/e421996c87d5f3c8f7eeadd502d4ad402c89464d/debian-openssl-1.1.x/query-engine to /workspace/node_modules/prisma/query-engine-debian-openssl-1.1.x
Error executing lifecycle: unexpected EO
```

(Currently trying to investigate if there is a way to fix things on Prisma side - but usually this is not part of the deployment process that fails so this is pretty new for us. Happy for input here what could be the cause.)

**Comment by mrkurt** [comment]
@janpio do you all have a Dockerfile to build an app like this? It's very strange that step is hanging with Heroku's buildpack setup. I'm curious if the same thing happens with raw docker.

**Comment by janpio** [comment]
The reproduction repo that @ryanflorence mentioned above https://github.com/ryanflorence/fly-prisma is a fine. It's really minimal.

**Comment by mrkurt** [comment]
@janpio oh yes that's what I'm using! But I want to try building that with just Docker, if possible.

**Comment by janpio** [comment]
Hm, I honestly would not know what to do besides cloning that repo in a Docker container environment :D Am I missing something here? What are you trying to achieve?

**Comment by janpio** [comment]
I tried to understand our code a bit further:

In the installation of the project dependencies, we have 2 dependencies that want to download binaries: `postinstall` of `@prisma/engines` and then later `@prisma/client` via `prisma generate`. Usually the first downloads the file successfully, putting them into a cache location along the way. And then the second one can just take them from the cache instead of accessing the network again.

And that is also what seems to be happening in the output we are seeing.

This is exactly where things go wrong:
https://github.com/prisma/prisma/blob/14ffd0093c1b8c7db1f45e9f04c5fab7c1c120d2/src/packages/fetch-engine/src/download.ts#L285-L293
The `copying` is still output, but then any of the further file commands somehow leads to this error `Error executing lifecycle: unexpected EOF`.

**Comment by mrkurt** [comment]
We think we figured out the issue. It wasn't really prisma related, it was a problem with high IO load and the encrypted volumes we use for remote Docker daemons. It seems to manifest when there's a lot of disk activity involved in a build. Heroku's buildpack + Node is worst case for this.

We should have a fix out today!

**Comment by mrkurt** [comment]
@janpio I just reset your builder, if it's simple to run again will you give it a shot and see how it goes?

**Comment by janpio** [comment]
Unfortunately not change when I use `flyctl deploy` or `flyctl launch`:

```
...


> @prisma/client@2.21.2 postinstall /workspace/node_modules/@prisma/client
> node scripts/postinstall.js

Environment variables loaded from prisma/.env
Prisma schema loaded from prisma/schema.prisma
Error executing lifecycle: unexpected EOF
```

**Comment by janpio** [comment]
I tried again with a new checkout of the repo from @ryanflorence today, and it seems to work... better but not completely yet:

```
C:\Users\Jan\Documents\throwaway
λ git clone https://github.com/ryanflorence/fly-prisma fly-prisma2
Cloning into 'fly-prisma2'...
remote: Enumerating objects: 24, done.
remote: Counting objects: 100% (24/24), done.
remote: Compressing objects: 100% (18/18), done.
remote: Total 24 (delta 7), reused 22 (delta 5), pack-reused 0
Receiving objects: 100% (24/24), 11.43 KiB | 2.86 MiB/s, done.
Resolving deltas: 100% (7/7), done.

C:\Users\Jan\Documents\throwaway
λ cd fly-prisma2

C:\Users\Jan\Documents\throwaway\fly-prisma2 (main -> origin) (script)
λ flyctl launch
Update available 0.0.210 -> 0.0.211
Update with flyctl.exe version update

Creating app in C:\Users\Jan\Documents\throwaway\fly-prisma2
An existing fly.toml file was found for app lively-dew-618
? Would you like to copy its configuration to the new app? Yes
Scanning source code
Detected Dockerfile app
Automatically selected personal organization: Jan Piotrowski
? Select region: fra (Frankfurt, Germany)
Created app misty-sun-7168 in organization personal
Wrote config file fly.toml
? Would you like to deploy now? Yes
Deploying misty-sun-7168
==> Validating app configuration
--> Validating app configuration done
Services
TCP 80/443 ⇢ 8080
WARN Error connecting to local docker daemon: error during connect: This error may indicate that the docker daemon is not running.: Get "http://%2F%2F.%2Fpipe%2Fdocker_engine/_ping": open //./pipe/docker_engine: The system cannot find the file specified.
Remote builder fly-builder-patient-wood-4676 ready
==> Creating build context
--> Creating build context done
==> Building image with Docker
Step 1/9 : FROM node:12.16.2-alpine
12.16.2-alpine: Pulling from library/node
cbdbe7a5bc2a: Pull complete
f98f8aade4b1: Pull complete
6c9c37ffd044: Pull complete
003e8eba0035: Pull complete
Digest: sha256:f48e315114006f6d94f7661b317156178fadd5a2b6aab060cb2eb2759fe3a65a
Status: Downloaded newer image for node:12.16.2-alpine
 ---> 4b3c4c2702d2
Step 2/9 : ENV LANG=C.UTF-8
 ---> Running in 233db3299b35
 ---> fa10d0c86459
Step 3/9 : ENV GLIBC_VERSION 2.30-r0
 ---> Running in cdafa66643ef
 ---> dd269d65a99b
Step 4/9 : RUN apk add --update curl &&   curl -Lo /etc/apk/keys/sgerrand.rsa.pub https://alpine-pkgs.sgerrand.com/sgerrand.rsa.pub &&   curl -Lo glibc.apk "https://github.com/sgerrand/alpine-pkg-glibc/releases/download/${GLIBC_VERSION}/glibc-${GLIBC_VERSION}.apk" &&   curl -Lo glibc-bin.apk "https://github.com/sgerrand/alpine-pkg-glibc/releases/download/${GLIBC_VERSION}/glibc-bin-${GLIBC_VERSION}.apk" &&   apk add glibc-bin.apk glibc.apk &&   /usr/glibc-compat/sbin/ldconfig /lib /usr/glibc-compat/lib &&   echo 'hosts: files mdns4_minimal [NOTFOUND=return] dns mdns4' >> /etc/nsswitch.conf &&   apk del curl &&   rm -rf glibc.apk glibc-bin.apk /var/cache/apk/*
 ---> Running in 7b18b55f0700
fetch http://dl-cdn.alpinelinux.org/alpine/v3.11/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.11/community/x86_64/APKINDEX.tar.gz
(1/4) Installing ca-certificates (20191127-r2)
(2/4) Installing nghttp2-libs (1.40.0-r1)
(3/4) Installing libcurl (7.67.0-r3)
(4/4) Installing curl (7.67.0-r3)
Executing busybox-1.31.1-r9.trigger
Executing ca-certificates-20191127-r2.trigger
OK: 9 MiB in 20 packages
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   451  100   451    0     0    803      0 --:--:-- --:--:-- --:--:--   803
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   638  100   638    0     0   3687      0 --:--:-- --:--:-- --:--:--  3687
100 4304k  100 4304k    0     0  6715k      0 --:--:-- --:--:-- --:--:-- 13.6M
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   642  100   642    0     0   3242      0 --:--:-- --:--:-- --:--:--  3242
100  980k  100  980k    0     0  2197k      0 --:--:-- --:--:-- --:--:-- 2197k
(1/2) Installing glibc (2.30-r0)
(2/2) Installing glibc-bin (2.30-r0)
Executing glibc-bin-2.30-r0.trigger
/usr/glibc-compat/sbin/ldconfig: /usr/glibc-compat/lib/ld-linux-x86-64.so.2 is not a symbolic link

OK: 20 MiB in 22 packages
/usr/glibc-compat/sbin/ldconfig: /usr/glibc-compat/lib/ld-linux-x86-64.so.2 is not a symbolic link

(1/4) Purging curl (7.67.0-r3)
(2/4) Purging libcurl (7.67.0-r3)
(3/4) Purging ca-certificates (20191127-r2)
Executing ca-certificates-20191127-r2.post-deinstall
(4/4) Purging nghttp2-libs (1.40.0-r1)
Executing busybox-1.31.1-r9.trigger
Executing glibc-bin-2.30-r0.trigger
/usr/glibc-compat/sbin/ldconfig: /usr/glibc-compat/lib/ld-linux-x86-64.so.2 is not a symbolic link

OK: 19 MiB in 18 packages
 ---> 1b4dddb2871b
Step 5/9 : COPY package.json .
 ---> 04983036e2db
Step 6/9 : COPY package-lock.json .
 ---> 8825f6b7bb1d
Step 7/9 : ADD . .
 ---> 6d13bffa4a5f
Step 8/9 : RUN npm install -g @prisma/client prisma && prisma generate
 ---> Running in a717afd17003

> prisma@2.21.2 preinstall /usr/local/lib/node_modules/prisma
> node scripts/preinstall-entry.js

/usr/local/bin/prisma2 -> /usr/local/lib/node_modules/prisma/build/index.js
/usr/local/bin/prisma -> /usr/local/lib/node_modules/prisma/build/index.js

> prisma@2.21.2 install /usr/local/lib/node_modules/prisma
> node scripts/install-entry.js


> @prisma/client@2.21.2 postinstall /usr/local/lib/node_modules/@prisma/client
> node scripts/postinstall.js

[Error: EACCES: permission denied, mkdir '/usr/local/lib/node_modules/.prisma'] {
  errno: -13,
  code: 'EACCES',
  syscall: 'mkdir',
  path: '/usr/local/lib/node_modules/.prisma'
}
Environment variables loaded from prisma/.env
Error: Can't write to /usr/local/lib/node_modules/prisma/node_modules/@prisma/engines please make sure you install "prisma" with the right permissions.

> @prisma/engines@2.21.0-36.e421996c87d5f3c8f7eeadd502d4ad402c89464d postinstall /usr/local/lib/node_modules/prisma/node_modules/@prisma/engines
> node download/index.js

+ prisma@2.21.2
+ @prisma/client@2.21.2
added 4 packages from 1 contributor in 11.278s
Environment variables loaded from prisma/.env
Prisma schema loaded from prisma/schema.prisma

> prisma@2.21.2 preinstall /prisma/node_modules/prisma
> node scripts/preinstall-entry.js


> prisma@2.21.2 install /prisma/node_modules/prisma
> node scripts/install-entry.js


> @prisma/engines@2.21.0-36.e421996c87d5f3c8f7eeadd502d4ad402c89464d postinstall /prisma/node_modules/@prisma/engines
> node download/index.js

npm WARN saveError ENOENT: no such file or directory, open '/prisma/package.json'
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN enoent ENOENT: no such file or directory, open '/prisma/package.json'
npm WARN prisma No description
npm WARN prisma No repository field.
npm WARN prisma No README data
npm WARN prisma No license field.

+ prisma@2.21.2
added 2 packages from 1 contributor and audited 2 packages in 1.192s
found 0 vulnerabilities


> @prisma/client@2.21.2 postinstall /prisma/node_modules/@prisma/client
> node scripts/postinstall.js

npm WARN saveError ENOENT: no such file or directory, open '/prisma/package.json'
npm WARN enoent ENOENT: no such file or directory, open '/prisma/package.json'
npm WARN prisma No description
npm WARN prisma No repository field.
npm WARN prisma No README data
npm WARN prisma No license field.

+ @prisma/client@2.21.2
added 2 packages from 1 contributor and audited 4 packages in 0.61s
found 0 vulnerabilities


✔ Installed the @prisma/client and prisma packages in your project

✔ Generated Prisma Client (2.21.2) to ./prisma/node_modules/@prisma/client in 108ms
You can now start using Prisma Client in your code. Reference: https://pris.ly/d/client
```
import { PrismaClient } from '@prisma/client'
const prisma = new PrismaClient()
```
 ---> aa1a6547f752
Step 9/9 : CMD ["node", "server.js"]
 ---> Running in e765b8c43cac
 ---> 08ac216262eb
Successfully built 08ac216262eb
Successfully tagged registry.fly.io/misty-sun-7168:deployment-1619723045
--> Building image done
==> Pushing image to fly
The push refers to repository [registry.fly.io/misty-sun-7168]
58a1631d6f2d: Pushed
4c5639bd2952: Pushed
677e47488b6d: Pushed
dfa06a413f5b: Pushed
84a5774bbf09: Pushed
c2606627a7db: Pushed
6b165f4bf4a2: Pushed
39e0c80c1842: Pushed
3e207b409db3: Pushed
deployment-1619723045: digest: sha256:41f9e7bc55ec8e536793308b8a8ae9abf42b97561d24df85925f61b7899b0a72 size: 2206
--> Pushing image done
Image: registry.fly.io/misty-sun-7168:deployment-1619723045
Image size: 569 MB
==> Creating release
Release v0 created

You can detach the terminal anytime without stopping the deployment
Monitoring Deployment

1 desired, 1 placed, 0 healthy, 1 unhealthy [restarts: 2] [health checks: 1 total]
v0 failed - Failed due to unhealthy allocations - no stable job version to auto revert to
Failed Instances

==> Failure #1

Instance
  ID            = 464becdd
  Version       = 0
  Region        = fra
  Desired       = run
  Status        = failed
  Health Checks = 1 total
  Restarts      = 2
  Created       = 16s ago

Recent Events
TIMESTAMP            TYPE            MESSAGE
2021-04-29T19:05:50Z Received        Task received by client
2021-04-29T19:05:50Z Task Setup      Building Task Directory
2021-04-29T19:06:05Z Started         Task started by client
2021-04-29T19:06:07Z Terminated      Exit Code: 1
2021-04-29T19:06:07Z Restarting      Task restarting in 1.177677011s
2021-04-29T19:06:09Z Started         Task started by client
2021-04-29T19:06:11Z Terminated      Exit Code: 1
2021-04-29T19:06:11Z Restarting      Task restarting in 1.036599267s
2021-04-29T19:06:13Z Started         Task started by client
2021-04-29T19:06:15Z Terminated      Exit Code: 1
2021-04-29T19:06:15Z Not Restarting  Exceeded allowed attempts 2 in interval 5m0s and mode is "fail"
2021-04-29T19:06:15Z Alloc Unhealthy Unhealthy because of failed task

Recent Logs
2021-04-29T19:06:12Z [info] Configuring virtual machine
2021-04-29T19:06:12Z [info] Pulling container image
2021-04-29T19:06:13Z [info] Unpacking image
2021-04-29T19:06:13Z [info] Preparing kernel init
2021-04-29T19:06:13Z [info] Configuring firecracker
2021-04-29T19:06:13Z [info] Starting virtual machine
2021-04-29T19:06:13Z [info] Starting init (commit: 665705e)...
2021-04-29T19:06:13Z [info] Running: `docker-entrypoint.sh node server.js` as root
2021-04-29T19:06:13Z [info] 2021/04/29 19:06:13 listening on [fdaa:0:23b9:a7b:67:464b:ecdd:2]:22 (DNS: [fdaa::3]:53)
2021-04-29T19:06:14Z [info] Error: Cannot find module '@prisma/client'
2021-04-29T19:06:14Z [info]   throw err;
2021-04-29T19:06:14Z [info]   ^
2021-04-29T19:06:14Z [info] internal/modules/cjs/loader.js:983
2021-04-29T19:06:14Z [info] - /server.js
2021-04-29T19:06:14Z [info] Require stack:
2021-04-29T19:06:14Z [info]     at Function.Module._load (internal/modules/cjs/loader.js:862:27)
2021-04-29T19:06:14Z [info]     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:980:15)
2021-04-29T19:06:14Z [info]     at require (internal/modules/cjs/helpers.js:77:18)
2021-04-29T19:06:14Z [info]     at Module.require (internal/modules/cjs/loader.js:1042:19)
2021-04-29T19:06:14Z [info]     at Module._compile (internal/modules/cjs/loader.js:1156:30)
2021-04-29T19:06:14Z [info]     at Object.<anonymous> (/server.js:1:26)
2021-04-29T19:06:14Z [info]     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1176:10)
2021-04-29T19:06:14Z [info]     at Module.load (internal/modules/cjs/loader.js:1000:32)
2021-04-29T19:06:14Z [info]     at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12) {
2021-04-29T19:06:14Z [info]     at Function.Module._load (internal/modules/cjs/loader.js:899:14)
2021-04-29T19:06:14Z [info] }
2021-04-29T19:06:14Z [info]   requireStack: [ '/server.js' ]
2021-04-29T19:06:14Z [info]   code: 'MODULE_NOT_FOUND',
2021-04-29T19:06:14Z [info] Main child exited normally with code: 1
2021-04-29T19:06:14Z [info] Starting clean up.
***v0 failed - Failed due to unhealthy allocations - no stable job version to auto revert to and deploying as v1

Troubleshooting guide at https://fly.io/docs/getting-started/troubleshooting/

C:\Users\Jan\Documents\throwaway\fly-prisma2 (main -> origin) (script)
λ
```

This now might indeed some problem with the project or Prisma - I will look into this again.

**Comment by janpio** [comment]
Ok, now we can see real problems happening:

This line `RUN npm install -g @prisma/client prisma && prisma generate` from `Dockerfile` does not make much sense in context of a Prisma project that already has these dependencies locally. They lead to these funny errors:

```
> @prisma/client@2.21.2 postinstall /usr/local/lib/node_modules/@prisma/client
> node scripts/postinstall.js

[Error: EACCES: permission denied, mkdir '/usr/local/lib/node_modules/.prisma'] {
  errno: -13,
  code: 'EACCES',
  syscall: 'mkdir',
  path: '/usr/local/lib/node_modules/.prisma'
}
Environment variables loaded from prisma/.env
Error: Can't write to /usr/local/lib/node_modules/prisma/node_modules/@prisma/engines please make sure you install "prisma" with the right permissions.

```

and

```
> @prisma/engines@2.21.0-36.e421996c87d5f3c8f7eeadd502d4ad402c89464d postinstall /prisma/node_modules/@prisma/engines
> node download/index.js

npm WARN saveError ENOENT: no such file or directory, open '/prisma/package.json'
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN enoent ENOENT: no such file or directory, open '/prisma/package.json'
npm WARN prisma No description
npm WARN prisma No repository field.
npm WARN prisma No README data
npm WARN prisma No license field.

+ prisma@2.21.2
added 2 packages from 1 contributor and audited 2 packages in 1.192s
found 0 vulnerabilities


> @prisma/client@2.21.2 postinstall /prisma/node_modules/@prisma/client
> node scripts/postinstall.js

npm WARN saveError ENOENT: no such file or directory, open '/prisma/package.json'
npm WARN enoent ENOENT: no such file or directory, open '/prisma/package.json'
npm WARN prisma No description
npm WARN prisma No repository field.
npm WARN prisma No README data
npm WARN prisma No license field.

+ @prisma/client@2.21.2
added 2 packages from 1 contributor and audited 4 packages in 0.61s
found 0 vulnerabilities
```

I'll try to remove that and see where we end up without.

**Comment by janpio** [comment]
With that line replaced with just `npm install`, we get a much better output:

```
C:\Users\Jan\Documents\throwaway\fly-prisma2>flyctl launch 
Update available 0.0.210 -> 0.0.211
Update with flyctl.exe version update

Creating app in C:\Users\Jan\Documents\throwaway\fly-prisma2
? Would you like to copy its configuration to the new app? Yes
Scanning source codecopy its configuration to the new app? (y/N) y
Detected Dockerfile app
Automatically selected personal organization: Jan Piotrowski
? Select region: lhr (London, United Kingdom)
Created app icy-cloud-3228 in organization personal
Wrote config file fly.toml
? Would you like to deploy now? Yes
Deploying icy-cloud-3228
==> Validating app configuration
--> Validating app configuration done
Services
TCP 80/443 ⇢ 8080
WARN Error connecting to local docker daemon: error during connect: This error may indicate that the docker daemon is not running.: Get "http://%2F%2F.%2Fpipe%2Fdocker_engine/_ping": open //./pipe/docker_engine: The system cannot find the file specified.
Remote builder fly-builder-patient-wood-4676 ready
==> Creating build context
--> Creating build context done
==> Building image with Docker
Step 1/9 : FROM node:12.16.2-alpine
 ---> 4b3c4c2702d2
Step 2/9 : ENV LANG=C.UTF-8
 ---> Using cache
 ---> fa10d0c86459
Step 3/9 : ENV GLIBC_VERSION 2.30-r0
 ---> Using cache
 ---> dd269d65a99b
Step 4/9 : RUN apk add --update curl &&   curl -Lo /etc/apk/keys/sgerrand.rsa.pub https://alpine-pkgs.sgerrand.com/sgerrand.rsa.pub &&   curl -Lo glibc.apk "https://github.com/sgerrand/alpine-pkg-glibc/releases/download/${GLIBC_VERSION}/glibc-${GLIBC_VERSION}.apk" &&   curl -Lo glibc-bin.apk "https://github.com/sgerrand/alpine-pkg-glibc/releases/download/${GLIBC_VERSION}/glibc-bin-${GLIBC_VERSION}.apk" &&   apk add glibc-bin.apk glibc.apk &&   /usr/glibc-compat/sbin/ldconfig /lib /usr/glibc-compat/lib &&   echo 'hosts: files mdns4_minimal [NOTFOUND=return] dns mdns4' >> /etc/nsswitch.conf &&   apk del curl &&   rm -rf glibc.apk glibc-bin.apk /var/cache/apk/*
 ---> Using cache
 ---> 1b4dddb2871b
Step 5/9 : COPY package.json .
 ---> Using cache
 ---> 7a3278a9a4b4
Step 6/9 : COPY package-lock.json .
 ---> Using cache
 ---> 5e1e5f0f27b7
Step 7/9 : ADD . .
 ---> 6ec815e3a1eb
Step 8/9 : RUN npm install
 ---> Running in 2ad1f96ef494
npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!

> prisma@2.21.2 preinstall /node_modules/prisma
> node scripts/preinstall-entry.js


> prisma@2.21.2 install /node_modules/prisma
> node scripts/install-entry.js


> @prisma/engines@2.21.0-36.e421996c87d5f3c8f7eeadd502d4ad402c89464d postinstall /node_modules/@prisma/engines
> node download/index.js


> @prisma/client@2.21.2 postinstall /node_modules/@prisma/client
> node scripts/postinstall.js

Environment variables loaded from prisma/.env
Prisma schema loaded from prisma/schema.prisma

✔ Generated Prisma Client (2.21.2) to ./node_modules/@prisma/client in 79ms
You can now start using Prisma Client in your code. Reference: https://pris.ly/d/client
```
import { PrismaClient } from '@prisma/client'
const prisma = new PrismaClient()
```
npm WARN script@ No repository field.

updated 64 packages and audited 64 packages in 6.63s
found 0 vulnerabilities

 ---> 0b09fe1446c0
Step 9/9 : CMD ["node", "server.js"]
 ---> Running in 689114c12d6d
 ---> c551e160f95b
Successfully built c551e160f95b
Successfully tagged registry.fly.io/icy-cloud-3228:deployment-1619724312
--> Building image done
==> Pushing image to fly
The push refers to repository [registry.fly.io/icy-cloud-3228]
15d414158718: Pushed
b462cd514dd0: Pushed
c999b9619a2e: Mounted from morning-sound-7515
4a02dc09111c: Mounted from morning-sound-7515
84a5774bbf09: Mounted from morning-sound-7515
c2606627a7db: Mounted from morning-sound-7515
6b165f4bf4a2: Mounted from morning-sound-7515
39e0c80c1842: Mounted from morning-sound-7515
3e207b409db3: Mounted from morning-sound-7515
deployment-1619724312: digest: sha256:efa1cfcb83487dff89d76a34be9681c6a07af89b10e8efcf2b5a47357810cede size: 2210
--> Pushing image done
Image: registry.fly.io/icy-cloud-3228:deployment-1619724312
Image size: 1.0 GB
==> Creating release
Release v0 created
```

That looks good for a normal install of a project using Prisma - the client is generated correctly.

Accessing the website still seems to kill the container though 🔥 If I interpret the logs correctly, it seems to restart every minute now and then crash again. I don't know enough about fly to see what is going on here.

**Comment by janpio** [comment]
@mrkurt To get around this confusion I tried to create a minimal project with Prisma myself that should show that fundamentally deployment to Fly works: https://github.com/janpio/fly-prisma-example

Unfortunately it seems you need to wiggle something first on that deployment maybe:

```
C:\Users\Jan\Documents\fly>flyctl launch
Update available 0.0.210 -> 0.0.211
Update with flyctl.exe version update

Creating app in C:\Users\Jan\Documents\fly
Scanning source code
Detected NodeJS app
Using the following build configuration:
        Builder: heroku/buildpacks:20
        Buildpacks:
Automatically selected personal organization: Jan Piotrowski
? Select region: lhr (London, United Kingdom)
Created app fragrant-brook-987 in organization personal
Wrote config file fly.toml
? Would you like to deploy now? No

C:\Users\Jan\Documents\fly>flyctl deploy
Update available 0.0.210 -> 0.0.211
Update with flyctl.exe version update

Deploying fragrant-brook-987
==> Validating app configuration
--> Validating app configuration done
Services
TCP 80/443 ⇢ 8080
WARN Error connecting to local docker daemon: error during connect: This error may indicate that the docker daemon is not running.: Get "http://%2F%2F.%2Fpipe%2Fdocker_engine/_ping": open //./pipe/docker_engine: The system cannot find the file specified.
Remote builder fly-builder-patient-wood-4676 ready
==> Building image with Buildpacks
20: Pulling from heroku/buildpacks
345e3491a907: Already exists
57671312ef6f: Already exists
5e9250ddb7d0: Already exists
8b2054040643: Already exists
c1c147505443: Already exists
6f1d1afc1550: Already exists
e2835300b80b: Already exists
2b47316a7aeb: Pull complete
ea9e04c443aa: Pull complete
8ac81ddb4cc0: Pull complete
1ade48917dfc: Pull complete
357fefdf9bc9: Pull complete
df2a11ebcd38: Pull complete
f564065db809: Pull complete
93efa6a4be6e: Pull complete
9a14857ac4ed: Pull complete
c10fd1b9aa75: Pull complete
5035dafe4b03: Pull complete
5232ada5c376: Pull complete
03fb7dba7baa: Pull complete
2f2c4a8a19b1: Pull complete
faf3b72e38e1: Pull complete
491c1ce9c4f9: Pull complete
16f6a3cc35a6: Pull complete
5c1bae75edc6: Pull complete
ca1ee74d6c50: Pull complete
ee8d3cc6086a: Pull complete
b26bc3ea84b6: Pull complete
7f712c8af484: Pull complete
5f5bec6da751: Pull complete
e3e2a82272a0: Pull complete
730d9d09549e: Pull complete
8ff46e742297: Pull complete
c2fdb02fdce9: Pull complete
6d04259a2c0b: Pull complete
d5a29f24c14a: Pull complete
9215aa0839f1: Pull complete
4f4fb700ef54: Pull complete
Digest: sha256:dfad52c21d16902d8bb9dfe42ce22c5e7b41ade32cfbec1f17a04adafc1ab71e
Status: Downloaded newer image for heroku/buildpacks:20
20: Pulling from heroku/pack
345e3491a907: Already exists
57671312ef6f: Already exists
5e9250ddb7d0: Already exists
8b2054040643: Already exists
c1c147505443: Already exists
d80b9b4d5954: Pull complete
2e7361bc96b4: Pull complete
Digest: sha256:4076ef6eac228aa79b166bde8bd8017ac7bcef4e7bdaf0a4fdc48828c3a9e8ee
Status: Downloaded newer image for heroku/pack:20
===> DETECTING
======== Output: heroku/ruby@0.0.1 ========
no
err:  heroku/ruby@0.0.1 (1)
err:  salesforce/nodejs-fn@2.0.6 (1)
err:  salesforce/nodejs-fn@2.0.6 (1)
Warning: Warning: buildpack heroku/jvm has a "version" key. This key is deprecated in build plan requirements in buildpack API 0.3. "metadata.version" should be used instead
3 of 4 buildpacks participating
heroku/nodejs-engine 0.7.3
heroku/nodejs-npm    0.4.3
heroku/procfile      0.6.2
===> ANALYZING
Previous image with name "registry.fly.io/fragrant-brook-987:deployment-1619725991" not found
===> RESTORING
===> BUILDING
[INFO] Node.js Buildpack
[INFO] Setting NODE_ENV to production
[INFO] Installing toolbox
[INFO] - yj

[Installing Node]
[INFO] Getting Node version
[INFO] Resolving Node version
[INFO] Downloading and extracting Node v12.22.1

[Parsing package.json]
[INFO] Parsing package.json
[INFO] Using npm v6.14.12 from Node
[INFO] Installing node modules from ./package-lock.json

> prisma@2.21.2 preinstall /workspace/node_modules/prisma
> node scripts/preinstall-entry.js


> @prisma/engines@2.21.0-36.e421996c87d5f3c8f7eeadd502d4ad402c89464d postinstall /workspace/node_modules/@prisma/engines
> node download/index.js


> prisma@2.21.2 install /workspace/node_modules/prisma
> node scripts/install-entry.js


> @prisma/client@2.21.2 postinstall /workspace/node_modules/@prisma/client
> node scripts/postinstall.js

Prisma schema loaded from prisma/schema.prisma
Error executing lifecycle: unexpected EOF
```
The initial error is back for that one.

**Comment by mrkurt** [comment]
@janpio looks like the build is working (sometimes), and the app is failing to pass health checks. I think prisma listens on `127.0.0.1` by default, it needs to be `0.0.0.0` for us to connect to it. That's a relatively normal problem, we just need to polish up our UX when it happens.

The EOF errors are things we're still chasing down. It turns out they happen for more than just the initial reason we fixed!

**Comment by janpio** [comment]
> I think prisma listens on `127.0.0.1` by default, it needs to be `0.0.0.0` for us to connect to it. That's a relatively normal problem, we just need to polish up our UX when it happens.

Hm interesting, that is not configurable even as this was never really a problem. We start a local binary that we then usually can directly connect to. 

Do you have any idea haw that puts the container in an unhealthy check? In theory the server in my repro should start just fine with no Prisma interaction - only if a connection comes in it would try to execute a query.

**Comment by mrkurt** [comment]
I phrased that badly! The example uses express, which needs to listen on 0.0.0.0. It's actually configured here: https://github.com/ryanflorence/fly-prisma/blob/main/server.js#L14

We health check the app port to determine whether we can send traffic to it. So the instances are coming up and running just fine, we just stop them because we're unable to connect. You can remove the entire `[[services]]` definition in your fly.toml to disable our external load balancer and turn off this check for simplicity.

**Comment by janpio** [comment]
Hm yeah, my clean reproduction unfortunately keeps getting `Error executing lifecycle: unexpected EOF` errors. 

I would really like to set up an end to end test on our side that deploying a simple web server returning some data with Prisma on Fly, optimally from a Postgres instance also running on Fly. Only way we can make sure it keeps working and we have something to point people like Ryan to in the future :D

**Comment by mrkurt** [comment]
The simple workaround is to install and run docker locally, then run `fly deploy --local-only` to force its use. This should work really well until we get our remote builder EOFs tracked down.

**Comment by janpio** [comment]
Ok, just ping me when this happens here and I will give our live repro another try.

**Comment by jeromegn** [comment]
I tweaked our remote builders quite a bit and was able to deploy this with both the buildpack and the Dockerfile from the example repo.

Until we change this for everybody, if you want to try the new builder you will have to destroy your current builder with `flyctl apps destroy fly-builder-...`. Then you can start a new build.

However, it looks like there's another issue? I don't think it's a build-time problem though.

```
2021-05-04T17:15:15Z [info] Configuring firecracker
2021-05-04T17:15:15Z [info] Starting virtual machine
2021-05-04T17:15:16Z [info] Starting init (commit: 665705e)...
2021-05-04T17:15:16Z [info] Running: `/cnb/process/web` as heroku
2021-05-04T17:15:16Z [info] 2021/05/04 17:15:16 listening on [fdaa:0:33:a7b:aa2:d682:40fe:2]:22 (DNS: [fdaa::3]:53)
2021-05-04T17:15:16Z [info]   ^
2021-05-04T17:15:16Z [info]   throw err;
2021-05-04T17:15:16Z [info] internal/modules/cjs/loader.js:883
2021-05-04T17:15:16Z [info] Error: Cannot find module '.prisma/client'
2021-05-04T17:15:16Z [info] Require stack:
2021-05-04T17:15:16Z [info] - /workspace/node_modules/@prisma/client/index.js
2021-05-04T17:15:16Z [info] - /workspace/server.js
2021-05-04T17:15:16Z [info]     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
2021-05-04T17:15:16Z [info]     at Function.Module._load (internal/modules/cjs/loader.js:725:27)
2021-05-04T17:15:16Z [info]     at Module.require (internal/modules/cjs/loader.js:952:19)
2021-05-04T17:15:16Z [info]     at require (internal/modules/cjs/helpers.js:88:18)
2021-05-04T17:15:16Z [info]     at Object.<anonymous> (/workspace/node_modules/@prisma/client/index.js:1:16)
2021-05-04T17:15:16Z [info]     at Module._compile (internal/modules/cjs/loader.js:1063:30)
2021-05-04T17:15:16Z [info]     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
2021-05-04T17:15:16Z [info]     at Module.load (internal/modules/cjs/loader.js:928:32)
2021-05-04T17:15:16Z [info]     at Function.Module._load (internal/modules/cjs/loader.js:769:14)
2021-05-04T17:15:16Z [info]     at Module.require (internal/modules/cjs/loader.js:952:19) {
2021-05-04T17:15:16Z [info]   requireStack: [
2021-05-04T17:15:16Z [info]   code: 'MODULE_NOT_FOUND',
2021-05-04T17:15:16Z [info]     '/workspace/server.js'
2021-05-04T17:15:16Z [info]     '/workspace/node_modules/@prisma/client/index.js',
2021-05-04T17:15:16Z [info] }
2021-05-04T17:15:16Z [info]   ]
2021-05-04T17:15:17Z [info] Main child exited normally with code: 1
2021-05-04T17:15:17Z [info] Starting clean up.
```

**Comment by janpio** [comment]
That points to something going wrong before. I will try this and report back.

**Comment by mrkurt** [comment]
@janpio one other thing! We're just building with Heroku's buildpack right now. If there's a better way to package a Prisma app we can drop the buildpack and give people a better experience. Let us know if you all think of anything in this area.

**Comment by janpio** [comment]
Thanks for the effort @jeromegn, but even after destroying the `fly-builder-...` I am seeing the old error still:
https://github.com/janpio/fly-prisma-example

```
C:\Users\Jan\Documents\fly>flyctl destroy fly-builder-patient-wood-4676
Update available 0.0.210 -> 0.0.212
Update with flyctl.exe version update

Destroying an app is not reversible.
? Destroy app fly-builder-patient-wood-4676? Yes
Destroyed app fly-builder-patient-wood-4676

C:\Users\Jan\Documents\fly>flyctl launch
Update available 0.0.210 -> 0.0.212
Update with flyctl.exe version update

Creating app in C:\Users\Jan\Documents\fly
An existing fly.toml file was found for app wild-grass-7529
? Would you like to copy its configuration to the new app? Yes
Scanning source code
Detected NodeJS app
Using the following build configuration:
        Builder: heroku/buildpacks:20
        Buildpacks:
Automatically selected personal organization: Jan Piotrowski
? Select region: lhr (London, United Kingdom)
Created app winter-meadow-5740 in organization personal
Wrote config file fly.toml
? Would you like to deploy now? No

C:\Users\Jan\Documents\fly>flyctl deploy
Update available 0.0.210 -> 0.0.212
Update with flyctl.exe version update

Deploying winter-meadow-5740
==> Validating app configuration
--> Validating app configuration done
Services
TCP 80/443 ⇢ 8080
WARN Error connecting to local docker daemon: error during connect: This error may indicate that the docker daemon is not running.: Get "http://%2F%2F.%2Fpipe%2Fdocker_engine/_ping": open //./pipe/docker_engine: The system cannot find the file specified.
Remote builder fly-builder-frosty-wave-2138 ready
==> Building image with Buildpacks
20: Pulling from heroku/buildpacks
345e3491a907: Pulling fs layer
57671312ef6f: Pulling fs layer
5e9250ddb7d0: Pulling fs layer
8b2054040643: Pulling fs layer
c1c147505443: Pulling fs layer
6f1d1afc1550: Pulling fs layer
e2835300b80b: Pulling fs layer
19bc210ea35e: Pull complete
9b417b241dd8: Pull complete
9df61bb30393: Pull complete
c2391fee192e: Pull complete
357fefdf9bc9: Pull complete
df2a11ebcd38: Pull complete
f564065db809: Pull complete
93efa6a4be6e: Pull complete
9a14857ac4ed: Pull complete
c10fd1b9aa75: Pull complete
5035dafe4b03: Pull complete
5232ada5c376: Pull complete
03fb7dba7baa: Pull complete
2f2c4a8a19b1: Pull complete
faf3b72e38e1: Pull complete
491c1ce9c4f9: Pull complete
63a7d1f24850: Pull complete
1f485bfd4d74: Pull complete
7a9727666e95: Pull complete
376558cb7b07: Pull complete
b26bc3ea84b6: Pull complete
8ff46e742297: Pull complete
c2fdb02fdce9: Pull complete
ee8d3cc6086a: Pull complete
7f712c8af484: Pull complete
5f5bec6da751: Pull complete
e3e2a82272a0: Pull complete
730d9d09549e: Pull complete
6acaf84b72df: Pull complete
d5a29f24c14a: Pull complete
9215aa0839f1: Pull complete
4f4fb700ef54: Pull complete
Digest: sha256:741d524df2056caa44b36b61c1c09ca41e027634b74bd5f230f78f1194b4817b
Status: Downloaded newer image for heroku/buildpacks:20
20: Pulling from heroku/pack
345e3491a907: Already exists
57671312ef6f: Already exists
5e9250ddb7d0: Already exists
8b2054040643: Already exists 
c1c147505443: Already exists
fe6892fe36bc: Pull complete
97339a0e4c57: Pull complete
Digest: sha256:5290610edd1e87dc292b1c518c88309300b32c18b75cdd2676d1230ef1b7d9b9
Status: Downloaded newer image for heroku/pack:20
===> DETECTING
======== Output: heroku/ruby@0.0.1 ========
no
err:  heroku/ruby@0.0.1 (1)
err:  salesforce/nodejs-fn@2.0.6 (1)
err:  salesforce/nodejs-fn@2.0.6 (1)
Warning: Warning: buildpack heroku/jvm has a "version" key. This key is deprecated in build plan requirements in buildpack API 0.3. "metadata.version" should be used instead
3 of 4 buildpacks participating
heroku/nodejs-engine 0.7.3
heroku/nodejs-npm    0.4.3
heroku/procfile      0.6.2
===> ANALYZING
Previous image with name "registry.fly.io/winter-meadow-5740:deployment-1620164816" not found
===> RESTORING
===> BUILDING
[INFO] Node.js Buildpack
[INFO] Setting NODE_ENV to production
[INFO] Installing toolbox
[INFO] - yj

[Installing Node]
[INFO] Getting Node version
[INFO] Resolving Node version
[INFO] Downloading and extracting Node v12.22.1

[Parsing package.json]
[INFO] Parsing package.json
[INFO] Using npm v6.14.12 from Node
[INFO] Installing node modules from ./package-lock.json

> prisma@2.21.2 preinstall /workspace/node_modules/prisma
> node scripts/preinstall-entry.js


> @prisma/engines@2.21.0-36.e421996c87d5f3c8f7eeadd502d4ad402c89464d postinstall /workspace/node_modules/@prisma/engines
> node download/index.js


> prisma@2.21.2 install /workspace/node_modules/prisma
> node scripts/install-entry.js


> @prisma/client@2.21.2 postinstall /workspace/node_modules/@prisma/client
> node scripts/postinstall.js

Prisma schema loaded from prisma/schema.prisma
Error executing lifecycle: unexpected EOF
```
(New builder is `fly-builder-frosty-wave-2138`)

@mrkurt _Any_ Node environment should work fine - Prisma is just a library that is embedded in an application, and does not really care much about the rest. (Unless the environment does not execute the `postinstall` scripts fully, that makes it unhappy)

**Comment by jeromegn** [comment]
Ah, I think this is different from the other issue! I'll keep looking into it. I think this is the idle timeout from our proxy being reached on the connection.

I was testing with

```toml
[build.args]
PRISMA_SKIP_POSTINSTALL_GENERATE = true
```

Without I also see the EOF.

**Comment by janpio** [comment]
> I was testing with
> 
> ```toml
> [build.args]
> PRISMA_SKIP_POSTINSTALL_GENERATE = true
> ```
> 
> Without I also see the EOF.

Then also your follow up error message `Error: Cannot find module '.prisma/client'` you posted makes sense: This is created during that generation. Running `npx prisma generate` should fix it, but that will of course probably also fail. It downloads a binary from S3 and puts it into locations.

**Comment by jeromegn** [comment]
Adding `DEBUG="*"` to the build args now shows it hangs here:

```
May 05 10:31:46 Prisma schema loaded from prisma/schema.prisma
May 05 10:31:46 2021-05-05T14:31:46.525Z prisma:download copying /home/heroku/.cache/prisma/master/e421996c87d5f3c8f7eeadd502d4ad402c89464d/debian-openssl-1.1.x/query-engine to /workspace/node_modules/prisma/query-engine-debian-openssl-1.1.x
```

**Comment by janpio** [comment]
Yep, similar state we were at before: https://github.com/superfly/flyctl/issues/405#issuecomment-826861737

**Comment by jeromegn** [comment]
Update: We've been able to successfully build this _sometimes_.

The issue is really with that step (the copying of `query-engine-debian-openssl-1.1.x`) with the buildpack. I do not know why, I don't really understand why it would be a problem and why it would sometimes work fast, other times slow and most times not at all.

I've tested this on Docker for Mac using the `pack` CLI:

```shell
# while in the fly-prisma directory
pack build prismatest --builder heroku/buildpacks:20
```

(if you'd like to try that, it might help to have extra datapoints)

First time I got a segfault from qemu, the next time I also got a segfault, but at a later stage. This might be because I'm on a M1 mac.

I've also ran the same command on my linux machine at home. I canceled the build after being stuck on the aforementioned step for over 5 minutes.

If building with Docker works well, then maybe that's the way forward for prisma projects 😕 (As mentioned by @mrkurt in https://github.com/superfly/flyctl/issues/405#issuecomment-832149108)

Oddly, Heroku can build this fine. I do not know how different the CNB is from their own buildpack.

**Comment by janpio** [comment]
Thanks for pinpointing this - we will continue our own investigation over at https://github.com/prisma/prisma/issues/6779#issuecomment-833529613 then.

What a weird bug. Let's see if we can reproduce it even.

**Comment by jeromegn** [comment]
I updated the original prisma issue, but I believe this should now be fixed. A kernel upgrade for our remote builders was necessary.

**Comment by janpio** [comment]
It works for me now - thanks so much for staying on this @jeromegn.

Here is my update (and close) of the Prisma side issue: https://github.com/prisma/prisma/issues/6779#issuecomment-834649804
